extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java

```diff
@@ -49,27 +41,35 @@ public YoutubePlaylistExtractor(StreamingService service, ListLinkHandler linkHa
     }
 
     @Override
-    public void onFetchPage(@Nonnull Downloader downloader) throws IOException, ExtractionException {
-        final String url = getUrl() + "&pbj=1";
-
-        initialAjaxJson = getJsonResponse(url, getExtractorLocalization());
-
-        initialData = initialAjaxJson.getObject(1).getObject("response");
+    public void onFetchPage(@Nonnull final Downloader downloader) throws IOException,
+            ExtractionException {
+        final Localization localization = getExtractorLocalization();
+        final byte[] body = JsonWriter.string(prepareDesktopJsonBuilder(localization,
+                getExtractorContentCountry())
+                .value("browseId", "VL" + getId())
+                .value("params", "wgYCCAA%3D") // Show unavailable videos
```

Comment:
Are we sure about this? What is the purpose of showing unavailable videos in playlists? YouTube does not show them normally, and in NewPipe they would just create problems. Anyway, we'll think about this later.

Reply:
Before a YouTube update, they were shown every time, so I thought it may be useful for some users (premieres, temporary georestrictions (think to music releases), ...).

Reply:
There was a button on YouTube to show them manually if you were logged in iirc.

Reply:
On a technical side, this should be just base64 URL-encoded protobuf.

Reply:
You don't have to be logged in to show them manually: it also works if you are a guest.